dev_guide/app_tutorials/jenkins_pipeline.adoc

```diff
@@ -0,0 +1,185 @@
+
+For this example, we are going to create a Jenkins Pipeline that will build, deploy, and verify a Node.js/MongoDB
+application using the https://github.com/openshift/nodejs-ex/blob/master/openshift/templates/nodejs-mongodb.json[nodejs-mongodb.json] template.
+
+[[creating-the-jenkins-master]]
+
+== Creating the Jenkins Master
+
+Creating the Jenkins master via the command line:
+
+----
+  // select the project that you want to use
+  // or create a new project with oc new-project <project_name>
+  $ oc project <project_name>
+  // if you want to use persistent storage, use jenkins-persistent instead
+  $ oc new-app jenkins-ephemeral
+----
+
+[NOTE]
+====
+If Jenkins auto-provisioning is enabled on your cluster, and you don't need to make any customizations to the
+Jenkins master, you can skip the above step. For more information about Jenkins autoprovisioning see xref:../../install_config/configuring_pipeline_execution.html[Configuring Pipeline Execution].
+====
+
+[[the-pipeline-build-config]]
+
+== The Pipeline Build Config
+
+Now that the Jenkins master is up and running, we will create a build config that employs the Jenkins pipeline strategy
+to build, deploy, and scale our Node.js/MongoDB example application.
+
+Create a file named *nodejs-sample-pipeline.yaml* with the below content:
+
+[source,yaml]
+----
+kind: "BuildConfig"
+apiVersion: "v1"
+metadata:
+  name: "nodejs-sample-pipeline"
+spec:
+  strategy:
+    jenkinsPipelineStrategy:
+      jenkinsfile: <pipeline content from below>
+    type: JenkinsPipeline
+----
+
+For more information about configuring the Pipeline Build Strategy see xref:../../builds/build_strategies.html#pipeline-strategy-options[Pipeline Strategy Options].
+
+[[the-jenkinsfile]]
+
+== The Jenkinsfile
+
+Now that we have created the build config with a jenkinsPipelineStrategy, we need to tell the pipeline what to do.
+We will do this by using an inline Jenkinsfile, since we aren't going to be creating our own git repository for this application.
+
+Below is the Jenkinsfile content which is written in groovy using the OpenShift DSL.  We will inline
+this content into the build config using the http://www.yaml.org/spec/1.2/spec.html#id2795688[YAML Literal Style].
+
+The completed build config can be viewed in the OpenShift Origin repository in the examples directory: https://github.com/openshift/origin/tree/master/examples/jenkins/pipeline/nodejs-sample-pipeline.yaml[nodejs-sample-pipeline.yaml].
+
+[source, groovy]
+----
+// path of the template to use
+def templatePath = 'https://raw.githubusercontent.com/openshift/nodejs-ex/master/openshift/templates/nodejs-mongodb.json'
+// name of the template that will be created
+def templateName = 'nodejs-mongodb-example'
+
+// timeout in 20 minutes if the pipeline hasn't completed
+timeout(time: 20, unit: 'MINUTES') {
+  // use a nodejs jenkins slave
+  node('nodejs') {
```

Comment:
This comment could just as easily have been made in the sample PR (and I may make it there as well) ...I mentioned this briefly in an email exchange I believe, maybe in the card ... but certainly reasonable that it could get lost in the shuffle.

The "declarative pipeline syntax" was introduced earlier this year:  https://jenkins.io/doc/book/pipeline/syntax/#declarative-pipeline

We've pulled in the necessary plugins, versions, to leverage it.  And we have already received questions as to whether we support, do we have examples, etc.

It could be as simple as duplicating all the good stuff @coreydaley has in this sample within the skeleton of declarative pipeline.  Leveraging the `pipeline` and `agent` most notably.

Thoughts?

Reply:
I can update to use the declarative pipeline, i'll update the other pr with the same.

Reply:
ok cool ... and yeah, just to be clear, by update I was thinking both forms ... but I'm on the fence there .... perhaps if it is the same core example that is redundant

Reply:
updated in next push to use declarative syntax

Reply:
let's pick the form we want people to use primarily.  if that's declarative, let's just show that...

Reply:
ok sounds good... declarative "is the future"  ... which I believe @coreydaley has ready to push

in lieu of both forms, and with going to declarative, let's put a footnote somewhere that says we also work with the predecessor, which is now referred to as "scripted pipeline"